odl/tomo/geometry/geometry.py

```diff
@@ -251,64 +375,139 @@ class DivergentBeamGeometry(Geometry):
 
-    A divergent beam geometry is characterized by the presence of a
-    point source.
+    A geometry characterized by the presence of a point-like ray source.
 
-    Special cases include fan beam in 2d and cone beam in 3d.
+    In 2D such a geometry is usually called "fan beam geometry", while
+    in 3D one speaks of "cone beam geometries".
     """
 
-    def src_position(self, mpar):
+    def src_position(self, mparam):
         """Source position function.
 
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter for which to calculate the source position.
+        mparam : `array-like`
+            Motion parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
 
         Returns
         -------
-        pos : `numpy.ndarray` (shape (`ndim`,))
-            Source position, an `ndim`-dimensional vector.
+        pos : `numpy.ndarray`, shape (ndim,) or (num_mparams, ndim)
+            Vector(s) pointing from the origin to the source.
+            If ``mparam`` is a single parameter, a single vector
+            is returned, otherwise a stack of vectors along axis 0.
         """
         raise NotImplementedError('abstract method')
 
-    def det_to_src(self, mpar, dpar, normalized=True):
-        """Vector pointing from a detector location to the source.
+    def det_to_src(self, angle, dparam, normalized=True):
+        """Vector or direction from a detector location to the source.
 
-        A function of the motion and detector parameters.
+        The unnormalized version of this vector is computed as follows::
 
-        The default implementation uses the `det_point_position` and
-        `src_position` functions. Implementations can override this, for
-        example if no source position is given.
+            vec = src_position(angle) - det_point_position(angle, dparam)
 
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter at which to evaluate.
-        dpar : `det_params` element
-            Detector parameter at which to evaluate.
-        normalized : bool, optional
-            If ``True``, return a normalized (unit) vector.
+        angle : `array-like`
+            One or several (Euler) angles in radians at which to
+            evaluate. An array should stack parameters along axis 0.
+        dparam : `det_params` element or `array-like`
+            Detector parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
 
         Returns
         -------
-        vec : `numpy.ndarray`, shape (`ndim`,)
-            (Unit) vector pointing from the detector to the source.
+        det_to_src : `numpy.ndarray`
+            Vector(s) pointing from a detector point to the source (at
+            infinity).
+            The shape of the returned array is as follows:
+
+            - ``angle`` and ``dparam`` single: ``(ndim,)``
```

Comment:
Not quite sure I like these cases all over the place, feels error prone.

What about instead saying that the returned array always has shape

```
(angle + dparam).shape + (ndim,)
```

that should cover all special cases and also allows a use case that this does not (i.e. a list of angles and dparams, without outer product) and also does not enforce any order. Users would call it by simply appending an axis where they want to use broadcasting.

This can be generalized to all of these, and would make the doc somewhat easier to read.